src/NuGetForUnity/Editor/NugetApiClientV3.cs

```diff
@@ -0,0 +1,798 @@
+        public async Task DownloadNupkgToFile(NugetPackageSourceV3 packageSource, INugetPackageIdentifier package, string outputFilePath)
+        {
+            var version = package.Version.ToLowerInvariant();
+            var id = package.Id.ToLowerInvariant();
+            using (var request = new HttpRequestMessage(HttpMethod.Get, $"{packageBaseAddress}{id}/{version}/{id}.{version}.nupkg"))
+            {
+                AddHeadersToRequest(request, packageSource);
+                using (var response = await httpClient.SendAsync(request).ConfigureAwait(false))
+                {
+                    await EnsureResponseIsSuccess(response).ConfigureAwait(false);
+                    using (var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false))
+                    {
+                        using (var fileStream = File.Create(outputFilePath))
+                        {
+                            await stream.CopyToAsync(fileStream).ConfigureAwait(false);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Fetches the package details from the server.
+        /// </summary>
+        /// <param name="packageSource">The package source that owns this client.</param>
+        /// <param name="package">The package to receive the details for.</param>
+        /// <param name="cancellationToken">
+        ///     Token to cancel the request.
+        /// </param>
+        /// <returns>
+        ///     The package details or null if the package is not found.
+        /// </returns>
+        public async Task<List<NugetFrameworkGroup>> GetPackageDetails(
+            NugetPackageSourceV3 packageSource,
+            INugetPackageIdentifier package,
+            CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrEmpty(registrationsBaseUrl))
+            {
+                Debug.LogError(
+                    $"There are no {nameof(registrationsBaseUrl)} specified in the API '{apiIndexJsonUrl}' so we can't receive package details.");
+                return null;
+            }
+
+            var responseString = await GetStringFromServerAsync(
+                    packageSource,
+                    $"{registrationsBaseUrl}{package.Id.ToLowerInvariant()}/index.json",
+                    cancellationToken)
+                .ConfigureAwait(false);
+            var registrationResponse = JsonUtility.FromJson<RegistrationResponse>(responseString.Replace(@"""@id"":", @"""atId"":"));
+
+            // without a version specified, the latest version is returned
+            var getLatestVersion = string.IsNullOrEmpty(package.Version);
+            var item = getLatestVersion ?
+                registrationResponse.items.OrderByDescending(registrationItem => new NugetPackageVersion(registrationItem.lower)).First() :
+                registrationResponse.items.FirstOrDefault(
```

Comment:
Until shortly I was a fan of preferring the Linq methods over the List ones to there is no mix with Linq and non Linq code. But now that even analyzers recommend using the List versions, because of the performance difference I switched.